pkg/backends/tcp_test.go

```diff
@@ -7,11 +7,122 @@ import (
 	"reflect"
 	"sync"
 	"testing"
+	"time"
 
 	"github.com/ansible/receptor/pkg/logger"
-	"github.com/ansible/receptor/pkg/netceptor"
 )
 
+func TestTCPDialer_GetAddr(t *testing.T) {
+	tests := []struct {
+		name    string
+		address string
+	}{
+		{
+			name:    "IPv4 address",
+			address: "127.0.0.1:8080",
+		},
+		{
+			name:    "IPv6 address",
+			address: "[::1]:8080",
+		},
+		{
+			name:    "Hostname",
+			address: "example.com:443",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b, err := NewTCPDialer(tt.address, true, nil, nil)
+			if err != nil {
+				t.Errorf("NewTCPDialer() error = %v", err)
+
+				return
+			}
+			if got := b.GetAddr(); got != tt.address {
+				t.Errorf("TCPDialer.GetAddr() = %v, want %v", got, tt.address)
+			}
+		})
+	}
+}
+
+func TestTCPDialer_GetTLS(t *testing.T) {
+	tlsConfig := &tls.Config{
+		MinVersion: tls.VersionTLS12,
+	}
+
+	tests := []struct {
+		name string
+		tls  *tls.Config
+	}{
+		{
+			name: "No TLS",
+			tls:  nil,
+		},
+		{
+			name: "With TLS config",
+			tls:  tlsConfig,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b, err := NewTCPDialer("127.0.0.1:8080", true, tt.tls, nil)
+			if err != nil {
+				t.Errorf("NewTCPDialer() error = %v", err)
+
+				return
+			}
+			if got := b.GetTLS(); !reflect.DeepEqual(got, tt.tls) {
+				t.Errorf("TCPDialer.GetTLS() = %v, want %v", got, tt.tls)
+			}
+		})
+	}
+}
+
+func TestTCPDialerStart(t *testing.T) {
+	tests := []struct {
+		name    string
+		address string
+		redial  bool
+		tls     *tls.Config
+		wantErr bool
+	}{
+		{
+			name:    "Successful start",
+			address: "127.0.0.1:9998",
```

Comment:
```suggestion
			address: "127.0.0.1:0",
```
this could fail intermittently, if you specify 0 you let the OS choose a port, which is more resilient.